crates/router/src/core/errors/utils.rs

```diff
@@ -10,6 +10,41 @@ pub trait StorageErrorExt<T, E> {
     fn to_duplicate_response(self, duplicate_response: E) -> error_stack::Result<T, E>;
 }
 
+impl<T> StorageErrorExt<T, errors::CustomersErrorResponse>
+    for error_stack::Result<T, errors::StorageError>
+{
+    #[track_caller]
+    fn to_not_found_response(
+        self,
+        not_found_response: errors::CustomersErrorResponse,
+    ) -> error_stack::Result<T, errors::CustomersErrorResponse> {
+        self.map_err(|err| {
+            if err.current_context().is_db_not_found() {
+                return err.change_context(not_found_response);
+            };
+            match err.current_context() {
+                errors::StorageError::CustomerRedacted => {
+                    err.change_context(errors::CustomersErrorResponse::CustomerRedacted)
+                }
+                _ => err.change_context(errors::CustomersErrorResponse::InternalServerError),
+            }
```

Comment:
```suggestion
            match err.current_context() {
                error if error.is_db_not_found() => err.change_context(not_found_response),
                errors::StorageError::CustomerRedacted => {
                    err.change_context(errors::CustomersErrorResponse::CustomerRedacted)
                }
                _ => err.change_context(errors::CustomersErrorResponse::InternalServerError),
            }
```

A similar thing should be possible here as well.